scripts/increment_auto_off.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# A program that ticks a counter down until deactivating log_persistence
+
+import logging
+import sys
+import json
+import time
+
+import requests
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+sh = logging.StreamHandler(sys.stdout)
+logger.addHandler(sh)
+
+loop = True
+
+while loop:
+  try:
+    response = requests.get('http://localhost:5001/settings/persist_logs', timeout=10)
+    state = response.json()
+
+    if state["persist_logs"] and state["auto_off_delay"] > 0:
+      body: dict
+      if state["auto_off_delay"] >= 1:
+        delay = state["auto_off_delay"] - 1
+        body = {
+          "persist_logs": True,
+          "auto_off_delay": delay,
+        }
+        logging.info(f"Persist logs will be automatically turned off in {delay} day(s)")
+      else:
+        body = {
+          "persist_logs": False,
+          "auto_off_delay": 14,  # Reset to default for next time
+        }
+        logging.info("Persist logs has been turned off automatically")
+
+      json_data = json.dumps(body)
+      response = requests.post(
+        url='http://localhost:5001/settings/persist_logs',
+        headers={'Content-Type': 'application/json'},
+        data=json_data,
+        timeout=10,
+      )
+
+      if response.ok:
+        loop = False
+  except Exception as exc:
+    logger.exception(f"Auto off increment has failed with the following exception, trying again in 5 seconds: {exc}")
+  finally:
+    time.sleep(5)  # Loop until there is a response.ok at the end
```

Comment:
The only mechanism by which this can fail is network issues, so I think that a simple retry will suffice
There is no way to have a key issue as the file is verified as being there at the start of every endpoint, and if it isn't there or doesn't have all the proper headers then the file is created or the headers added.